src/App.tsx

```diff
@@ -8,38 +8,106 @@ import { PostsList } from './components/PostsList';
+  const [selectedUser, setSelectedUser] = useState<null | User>(null);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+  const [selectedPost, setSelectedPost] = useState<null | Post>(null);
+  const [isError, setIsEroor] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useMemo(() => {
+    const fetchData = async () => {
+      if (selectedUser) {
+        try {
+          setIsLoading(true);
+          const response = await client.get<Post[]>(`/posts?userId=${selectedUser.id}`);
+
+          setUserPosts(response);
+        } catch {
+          setIsEroor(true);
+        } finally {
+          setIsLoading(false);
+        }
+      }
+    };
+
+    if (selectedUser) {
+      fetchData();
+    }
+  }, [selectedUser]);
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      try {
+        const response = await client.get<User[]>('/users');
+
+        setUsers(response);
+      } catch {
+        setIsEroor(true);
+      }
+    };
+
+    fetchUsers();
+  }, []);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  setSelectedPost={setSelectedPost}
+                  users={users}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                />
               </div>
 
-              <div className="block" data-cy="MainContent">
+              {!selectedUser && (
                 <p data-cy="NoSelectedUser">
                   No user selected
                 </p>
+              )}
 
-                <Loader />
+              {isLoading
+                ? <Loader />
+                : (
+                  <div className="block" data-cy="MainContent">
+                    {isError && (
+                      <div
+                        className="notification is-danger"
+                        data-cy="PostsLoadingError"
+                      >
+                        Something went wrong!
+                      </div>
+                    )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                    {selectedUser && userPosts.length === 0 && (
+                      <div
+                        className="notification is-warning"
+                        data-cy="NoPostsYet"
+                      >
+                        No posts yet
+                      </div>
+                    )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                    {userPosts.length !== 0 && (
```

Comment:
```suggestion
                    {userPosts.length && (
```

Reply:
Under this condition, 0 will be displayed if there are no posts